### DISCUSSION TITLE
SPECTRAL core and custom functions not getting referenced when running spectral lint command from CMD

### DISCUSSION THREAD

**Discussion by pankhiprasher** [stoplightio/spectral#1890] [Q&A]
Hey I have been raising requests and tickets on the stoplight support but haven't got any response yet.
In our project we are creating custom JS functions and already have core JS functions and we need to call it from CMD but when i run the spectral lint command , I  keep getting these errors - 
Also i need a way forward to have custom functions called in from my project structure where i have rulesets folder and respective js files and this testspectral all of them in rulesets folder and i run a command like - 
C:\Program Files>spectral lint --ruleset **rulesets**/testspectral.yaml C:\Users\xxx\API.yaml 

Any help would be really appreciated .

C:\Program Files>spectral lint --ruleset testspectral.yaml C:\Users\xxx\API.yaml 
Cannot find module 'c:/Program Files/functions/oasOp2xxResponse.js'
Require stack:
- C:\Users\xxx\AppData\Roaming\npm\node_modules\@stoplight\spectral-cli\dist\services\linter\utils\getRuleset.js
- C:\Users\xxx\AppData\Roaming\npm\node_modules\@stoplight\spectral-cli\dist\services\linter\utils\index.js
- C:\Users\xxx\AppData\Roaming\npm\node_modules\@stoplight\spectral-cli\dist\services\linter\linter.js
- C:\Users\xxx\AppData\Roaming\npm\node_modules\@stoplight\spectral-cli\dist\services\linter\index.js
- C:\Users\xxx\AppData\Roaming\npm\node_modules\@stoplight\spectral-cli\dist\commands\lint.js
- C:\Users\xxx\AppData\Roaming\npm\node_modules\@stoplight\spectral-cli\dist\index.js

**Comment by P0lip** [comment]
> In our project we are creating custom JS functions and already have core JS functions

How do you try to access these functions?

If you try to use `oasOp2xxResponse` from the OpenAPI ruleset, then it won't work because it's not intended to be used externally, besides that ruleset.

**Reply by pankhiprasher** [reply]
But if i am using lint command for the spectral file that has oasOp2xxResponse in the functions and i am using that function in one of my rules in that spectral file then , i would need it to be there when i am running it through command prompt from where i am running this - spectral lint --ruleset .spectral.yaml C:\Users\api-linting\API_1.0.yaml

**Reply by P0lip** [reply]
Can you show me the ruleset you have?
